CustomCollectionLayout/CustomCollectionViewLayout.swift

```diff
@@ -96,80 +96,50 @@ class CustomCollectionViewLayout: UICollectionViewLayout {
                     yOffset += itemSize.height
                 }
             }
-            if (self.itemAttributes == nil) {
-                self.itemAttributes = NSMutableArray(capacity: self.collectionView!.numberOfSections())
-            }
-            self.itemAttributes .addObject(sectionAttributes)
+
+            itemAttributes.append(sectionAttributes)
         }
-        
-        var attributes : UICollectionViewLayoutAttributes = self.itemAttributes.lastObject?.lastObject as! UICollectionViewLayoutAttributes
+      
+        let attributes = itemAttributes.last!.last!
         contentHeight = attributes.frame.origin.y + attributes.frame.size.height
-        self.contentSize = CGSizeMake(contentWidth, contentHeight)
+        contentSize = CGSizeMake(contentWidth, contentHeight)
     }
     
     override func collectionViewContentSize() -> CGSize {
-        return self.contentSize
+        return contentSize
     }
     
-    override func layoutAttributesForItemAtIndexPath(indexPath: NSIndexPath) -> UICollectionViewLayoutAttributes! {
-        return self.itemAttributes[indexPath.section][indexPath.row] as! UICollectionViewLayoutAttributes
+    override func layoutAttributesForItemAtIndexPath(indexPath: NSIndexPath) -> UICollectionViewLayoutAttributes? {
+        return itemAttributes[indexPath.section][indexPath.row]
     }
     
     override func layoutAttributesForElementsInRect(rect: CGRect) -> [UICollectionViewLayoutAttributes]? {
-        var attributes = [UICollectionViewLayoutAttributes]()
-        if self.itemAttributes != nil {
-            for section in self.itemAttributes {
-                
-                let filteredArray  =  section.filteredArrayUsingPredicate(
-                    
-                    NSPredicate(block: { (evaluatedObject, bindings) -> Bool in
-                        return CGRectIntersectsRect(rect, evaluatedObject.frame)
-                    })
-                    ) as! [UICollectionViewLayoutAttributes]
-                
-                
-                attributes.appendContentsOf(filteredArray)
-                
-            }
+      var attributes = [UICollectionViewLayoutAttributes]()
+      
+      for section in itemAttributes {
+        let filteredArray = section.filter { obj -> Bool in
+          return CGRectIntersectsRect(rect, obj.frame)
```

Comment:
The higher-order function `filter` does everything `filteredArrayUsingPredicate` does except it's easier to read.